### DISCUSSION TITLE
Regex Manager: Multiple Node Versions in one Dockerfile

### DISCUSSION THREAD

**Discussion by robertkowalski** [renovatebot/renovate#22759] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.60.0

### If you're self-hosting Renovate, select which platform you are using.

Github Enterprise Server

### Was this something which used to work for you, and then stopped?

It never worked

### Wanted end result.

We have a Dockerfile with multiple Node version set (16, 18, 20). We would like to update them with renovate, but so far no luck. 

PRs for 16 and 18 look fine, but in the PR for updating version 20 all other versions are also updated.

### What you tried so far.

```
# renovate: versioning=regex:^v16\.(?<minor>\d+)\.(?<patch>\d+)$ extractVersion=^v16\.(?<minor>\d+)\.(?<patch>\d+)$
ENV NODE_VERSION_16=v16.13.2
# renovate: versioning=regex:^v18\.(?<minor>\d+)\.(?<patch>\d+)$ extractVersion=^v18\.(?<minor>\d+)\.(?<patch>\d+)$
ENV NODE_VERSION_18=v18.0.0
# renovate: versioning=regex:^v20\.(?<minor>\d+)\.(?<patch>\d+)$ extractVersion=^v20\.(?<minor>\d+)\.(?<patch>\d+)$
ENV NODE_VERSION_20=v20.0.0
```

renovate.json:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "pinDigests": false,
  "regexManagers": [
    {
      "fileMatch": ["^Dockerfile$"],
      "datasourceTemplate": "github-tags",
      "depNameTemplate": "nodejs/node",
      "matchStrings": [
        "ARG NODE_VERSION=(?<currentValue>.*)\\s",
        "ENV NODE_VERSION_16=(?<currentValue>.*)\\s",
        "ENV NODE_VERSION_18=(?<currentValue>.*)\\s",
        "ENV NODE_VERSION_20=(?<currentValue>.*)\\s"
      ]
    }
  ]
}

```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
Sounds like you could set a package rule disabling major updates. Match it based on some of file name, manager, package name, etc

**Reply by robertkowalski** [reply]
that worked, many thanks!

```
  "packageRules": [
    {
      "matchPackageNames": ["nodejs/node"],
      "matchManagers": ["regex"],
      "matchUpdateTypes": ["major"],
      "enabled": false
    }
  ]
```
